share/spack/gitlab/cloud_pipelines/stacks/developer-tools-manylinux2014/spack.yaml

```diff
@@ -0,0 +1,89 @@
+spack:
+  view: false
+  packages:
+    all:
+      require: target=x86_64_v3
+  concretizer:
+    unify: true
+  definitions:
+  - default_specs:
+      # editors
+    - neovim~no_luajit
+    - py-pynvim
+    - emacs@29.1+json+native+treesitter   # note, pulls in gcc
+      # - tree-sitter is a dep, should also have cli but no package
+    - nano   # just in case
+      # tags and scope search helpers
+    - universal-ctags   # only maintained ctags, works better with c++
+    - direnv
+      # runtimes and compilers
+    - python
+    - llvm+link_llvm_dylib~lld~lldb~polly+python build_type=MinSizeRel   # for clangd, clang-format
+    - node-js   # for editor plugins etc., pyright language server
+    - npm
+    - cmake
+    - libtool
+    - go   # to build fzf, gh, hub
+    - rust+dev   # fd, ripgrep, hyperfine, exa, rust-analyzer
+    - binutils+ld+gold+plugins   # support linking with built gcc
+      # styling and lints
+    - astyle
+    - cppcheck
+    - uncrustify
+    - py-fprettify
+    - py-fortran-language-server
+    - py-python-lsp-server
+      # cli dev tools
+    - ripgrep
+    - gh
+    - fd
+    # - bfs # liburing: /usr/include/linux/ipv6.h:19:8: error: redefinition of 'struct in6_pktinfo'
+    - fzf
+    - tree
+    - jq
+    - py-yq
+    - hub
+    - ncdu
+    - eza
+    - lsd
+    - hyperfine
+    - htop
+    - tmux
+    - ccache
+      # ensure we can use a jobserver build and do this fast
+    - gmake
+    - ninja   # should be @kitware, can't be because of meson requirement
+    - openssl certs=system     # must be this, system external does not work
+    - libtree
+    - patchelf
+
+  - arch:
+    - '%gcc target=x86_64_v3'
+
+  specs:
+  - matrix:
+    - - $default_specs
+    - - $arch
+
+  compilers:
+  - compiler:
+      spec: gcc@=10.2.1
+      paths:
+        cc: /opt/rh/devtoolset-10/root/usr/bin/gcc
+        cxx: /opt/rh/devtoolset-10/root/usr/bin/g++
+        f77: /opt/rh/devtoolset-10/root/usr/bin/gfortran
+        fc: /opt/rh/devtoolset-10/root/usr/bin/gfortran
+      flags: {}
+      operating_system: centos7
+      target: x86_64
+      modules: []
+      environment: {}
+      extra_rpaths: []
+
+  ci:
+    pipeline-gen:
+    - build-job:
+        image: ecpe4s/manylinux2014:2024.03.11
```

Comment:
I think having this in ghcr would be better, but not required for a first pass on this.

Reply:
I can't push to the ghcr Spack registry -- can you mirror this image over there and I can update the reference?